src/components/RichLayout/index.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import Layout from 'components/Layout'
+import { LanguageFetcher } from 'components/Fetcher'
+import Header from './Header'
+import Navigation from 'Navigation'
+
+class HeaderAdapter extends React.Component {
+  static propTypes = {
+    location: PropTypes.string.isRequired
+  }
+
+  render () {
+    return <Header
+      languages={this.props.languages}
+      languageCallback={this.props.languageCallback}
+      currentLanguage={this.props.language}
+      navigation={new Navigation(this.props.location)}
+    />
+  }
+}
+
+class RichLayout extends React.Component {
+  static propTypes = {
+    className: PropTypes.string,
+    location: PropTypes.string.isRequired
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.gotoParent = this.gotoParent.bind(this)
+  }
+
+  gotoParent () {
+    // history.push(this.getParentPath()) fixme
+  }
+
+  render () {
+    return (<div>
+        <LanguageFetcher location={this.props.location} hideError={true} hideSpinner={true}>
+          <HeaderAdapter languageCallback={this.gotoParent} language={this.props.language} location={this.props.location}/>
+        </LanguageFetcher>
+
+        <Layout className={this.props.className}>
+          {this.props.children}
+        </Layout>
+      </div>
+    )
+  }
```

Comment:
This looks really cool. The only thing I don't like about this 'way' of fetching data is: For example, every time the language fetcher fetches data (because the user switched language) the header completely disappears for a second (since the fetcher loads the data in) and pops back in. I think, it would be even cooler to have the language-fetcher as parent element of the languageflyout, so only the components that really depend on the data are not displayed (another side effect would be that the "languages-data" wouldn't be passed through all the other elements like header-adapter, header, etc.